### PR TITLE
Deprecate prefix argument in read_csv and read_table

### LINKED ISSUE #43396: DEPR: prefix argument in read_csv/read_table
It conflicts with header and names I think and users should really just be using a list comprehension on the dataframe column names afterwards.

### PR DESCRIPTION
**Pull request by phofl**
- [x] closes #43396
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [x] whatsnew entry

cc @lithomas1 I am also on board with deprecating this. I don't think the keyword adds real value

### CODE REVIEW

File: doc/source/whatsnew/v1.4.0.rst [hunk 1]
```diff
@@ -517,6 +517,7 @@ Other Deprecations
...
 - Deprecated casting behavior when passing an item with mismatched-timezone to :meth:`DatetimeIndex.insert`, :meth:`DatetimeIndex.putmask`, :meth:`DatetimeIndex.where` :meth:`DatetimeIndex.fillna`, :meth:`Series.mask`, :meth:`Series.where`, :meth:`Series.fillna`, :meth:`Series.shift`, :meth:`Series.replace`, :meth:`Series.reindex` (and :class:`DataFrame` column analogues). In the past this has cast to object dtype. In a future version, these will cast the passed item to the index or series's timezone (:issue:`37605`)
 - Deprecated the 'errors' keyword argument in :meth:`Series.where`, :meth:`DataFrame.where`, :meth:`Series.mask`, and meth:`DataFrame.mask`; in a future version the argument will be removed (:issue:`44294`)
+- Deprecated the ``header`` keyword argument in :func:`read_csv` and :func:`read_table`, in a future version the argument will be removed (:issue:`43396`)
```
Comment (asishm) [inline doc/source/whatsnew/v1.4.0.rst:520]: shouldn't this be `prefix` (and not header)

**Comment by lithomas1** [comment]
Using ``parser.read_csv_check_warnings`` should fix the stacklevel issues. 
https://github.com/pandas-dev/pandas/blob/bdff2148ec5c81b2724ac77b3138d229994cffc4/pandas/tests/io/parser/conftest.py#L31-L39

It would be also nice to add an example to the user guide of how to add a prefix to the names.

File: doc/source/whatsnew/v1.4.0.rst (hunk 1 above)
Comment (phofl) [reply doc/source/whatsnew/v1.4.0.rst:520]: thx

**Comment by jreback** [comment]
thanks @phofl

**Pull request merged**